kernelci/cli/event.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2023 Collabora Limited
+# Author: Guillaume Tucker <guillaume.tucker@collabora.com>
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+"""Tool to interact with the Pub/Sub interface"""
+
+import sys
+import json
+
+import click
+
+import kernelci.api
+import kernelci.api.helper
+import kernelci.config
+from . import Args, kci
+
+
+@kci.group(name='event')
+def kci_event():
+    """Interact with Pub/Sub interface"""
+
+
+@kci_event.command(secrets=True)
+@click.argument('channel')
+@Args.config
+@Args.api
+def subscribe(config, api, channel, secrets):
+    """Subscribe to a Pub/Sub channel"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    sub_id = api.subscribe(channel)
+    click.echo(sub_id)
+
+
+@kci_event.command(secrets=True)
+@click.argument('sub_id')
+@Args.config
+@Args.api
+def unsubscribe(config, api, sub_id, secrets):
+    """Unsubscribe from a Pub/Sub channel"""
+    configs = kernelci.config.load(config)
+    api_config = configs['api'][api]
+    api = kernelci.api.get_api(api_config, secrets.api.token)
+    api.unsubscribe(sub_id)
+
+
+@kci_event.command(secrets=True)
+@click.option('--event-type',
+              help="CloudEvent type, api.kernelci.org by default")
+@click.option('--source',
+              help="CloudEvent source, https://api.kernelci.org/ by default")
```

Comment:
These should probably be coming from the API YAML config object.

Reply:
Do you mean adding arguments to `kernelci.cli.Args` class? But the class holds common command line arguments and these are specific to this command only.

Reply:
No I mean removing them as arguments.  There's already the `--api` argument and I think these things should be fixed for each API YAML config entry.

Reply:
Makes sense. We should let API set event type and source instead. I'll drop the arguments.

Reply:
We can also add fixed entries to TOML but it makes more sense if API sets the source and type of the events instead.